src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -517,4 +512,39 @@ final protected function applyCallbacks($value, $object, string $attribute, ?str
 
         return $callback ? $callback($value, $object, $attribute, $format, $context) : $value;
     }
+
+    /**
+     * Computes the normalization context merged with current one. Metadata always wins over global context, as more specific.
+     */
+    protected function getAttributeNormalizationContext(object $object, string $attribute, array $context): array
```

Comment:
Let's mark these protected methods `@internal` to ease us maintainability

Reply:
👍